### PR TITLE
(Fix) BBCode

### PR DESCRIPTION
**Pull request by Roardom**
Complete refactor of bbcode to allow identical nested elements (e.g. quotes inside of quotes).

Example of broken bbcode that currently breaks:

```
[quote=Test][list][*]Test Bullet 1[*]Test Bullet 2[/list][/quote]

[quote=Test][quote=Inside quote]Another test[/quote]Outside quote[/quote]

Fully outside
```

Before:
image

After:
image

### CODE REVIEW

File: app/Helpers/Bbcode.php [hunk 1]
```diff
@@ -7,359 +7,310 @@
...
+ * @author     Roardom <roardom@protonmail.com>
  * @license    https://www.gnu.org/licenses/agpl-3.0.en.html/ GNU Affero General Public License v3.0
  */
 
 namespace App\Helpers;
 
 class Bbcode
 {
-    public array $parsers = [
+    protected array $parsers = [
```
Comment (HDVinnie) [inline app/Helpers/Bbcode.php:18]: This can be private

File: app/Helpers/Bbcode.php [hunk 2]
```diff
@@ -7,359 +7,310 @@
...
-
-    /**
-     * Searches after a specified pattern and replaces it with provided structure.
-     */
-    protected function searchAndReplace(string $pattern, string $replace, string $source): ?string
-    {
-        while (\preg_match($pattern, (string) $source)) {
-            $source = \preg_replace($pattern, $replace, $source);
-        }
+        // Replace line breaks
```
Comment (HDVinnie) [inline app/Helpers/Bbcode.php:310]: These could be combined to one
```
        // Replace line breaks
        $source = \str_replace(["\r\n", "\n"], '<br>', $source);
```

File: app/Helpers/Bbcode.php [hunk 3]
```diff
@@ -7,359 +7,310 @@
...
+        );
+        $source = \preg_replace_callback(
+            '/\[img=(\d+)\](.*?)\[\/img\]/i',
+            fn ($matches) => '<img src="'.\htmlspecialchars($matches[2]).'" loading="lazy" width="'.$matches[1].'px">',
+            $source
+        );
+
+        // Youtube elements need to be replaced like this because the content inside the two tags
+        // has to be moved into an html attribute
+        $source = \preg_replace_callback(
```
Comment (HDVinnie) [inline app/Helpers/Bbcode.php:249]: Pretty sure `frameborder` is deprecated and obsolete. Can probably just remove them.

File: app/Helpers/Bbcode.php (hunk 1 above)
Comment (Roardom) [reply app/Helpers/Bbcode.php]: Done

File: app/Helpers/Bbcode.php (hunk 2 above)
Comment (Roardom) [reply app/Helpers/Bbcode.php:310]: Done

File: app/Helpers/Bbcode.php (hunk 3 above)
Comment (Roardom) [reply app/Helpers/Bbcode.php:249]: Done

**Pull request merged**